src/test/java/io/airlift/compress/zstd/UtilTest.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.airlift.compress.zstd;
+
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static io.airlift.compress.zstd.Util.get24BitLittleEndian;
+import static io.airlift.compress.zstd.Util.put24BitLittleEndian;
+import static org.testng.Assert.assertEquals;
+import static sun.misc.Unsafe.ARRAY_BYTE_BASE_OFFSET;
+
+public class UtilTest
```

Comment:
Call this `TestUtil`

Reply:
> Can you add a test to ensure the decompressor will throw a MalformedInputException if there are not enough bytes to read from the input buffer under the conditions addressed by this fix?

I'm not sure I can, the existing code was already checking the correct number of bytes, unfortunately the code then read too many bytes.

Reply:
Renamed `UtilTest` to `TestUtil`